### PR TITLE
Admin view of moderation action and report details

### PR DESCRIPTION
**Pull request by devinivy**
Two more methods for admin views here, for details on a moderation action (`com.atproto.admin.getModerationAction`) and on a moderation report (`com.atproto.admin.getModerationReport`).

### CODE REVIEW

File: packages/pds/src/api/com/atproto/admin/getModerationReport.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  server.com.atproto.admin.getModerationReport({
+    auth: ctx.adminVerifier,
+    handler: async ({ params }) => {
+      const { db, services } = ctx
+      const { id } = params
+      const moderationService = services.moderation(db)
+      const result = await moderationService.getReportOrThrow(id)
+      return {
+        encoding: 'application/json',
+        body: await moderationService.views.reportDetail(result),
```
Comment (dholms) [inline packages/pds/src/api/com/atproto/admin/getModerationReport.ts:14]: I quite like this syntax for views btw

File: packages/pds/src/services/moderation/views.ts [hunk 2]
```diff
@@ -2,24 +2,36 @@ import { Selectable } from 'kysely'
...
 import { ActorService } from '../actor'
+import { RecordService } from '../record'
 
 export class ModerationViews {
-  constructor(private db: Database) {}
+  constructor(private db: Database, private messageQueue: MessageQueue) {}
 
   services = {
     actor: ActorService.creator(),
+    record: RecordService.creator(this.messageQueue),
```
Comment (dholms) [inline packages/pds/src/services/moderation/views.ts:34]: the messageQueue dependency makes a good argument for splitting out a RecordView service. but probably not necessary in this PR

**Review by dholms — APPROVED** [review]
yup yup yup 👍

File: packages/pds/src/api/com/atproto/admin/getModerationReport.ts (hunk 1 above)
Comment (devinivy) [reply packages/pds/src/api/com/atproto/admin/getModerationReport.ts:14]: Agree, it's feelin good.

**Pull request merged**